src/DataCollector/SolrDataCollector.php

```diff
@@ -0,0 +1,238 @@
+            $time += $queryStruct['durationMs'];
+        }
+        $this->data = [
+            'queries' => $this->queries,
+            'totalTimeMs' => $time,
+        ];
+    }
+
+    public function getName(): string
+    {
+        return 'solr';
+    }
+
+    public function reset()
+    {
+        $this->data = [];
+    }
+
+    public function preExecuteRequest(SolariumPreExecuteRequestEvent $event): void
+    {
+        if (isset($this->currentRequest)) {
+            $this->failCurrentRequest();
+        }
+
+        $this->currentRequest = $event->getRequest();
+        $this->currentEndpoint = $event->getEndpoint();
+
+        $this->logger->debug($event->getEndpoint()->getBaseUri().$this->currentRequest->getUri());
+        $this->currentStartTime = microtime(true);
+    }
+
+    public function postExecuteRequest(SolariumPostExecuteRequestEvent $event): void
+    {
+        $endTime = microtime(true) - $this->currentStartTime;
+        if (!isset($this->currentRequest)) {
+            throw new \RuntimeException('Request not set');
+        }
+        if ($this->currentRequest !== $event->getRequest()) {
+            throw new \RuntimeException('Requests differ');
+        }
+
+        $this->log($event->getRequest(), $event->getResponse(), $event->getEndpoint(), $endTime);
+
+        $this->currentRequest = null;
+        $this->currentStartTime = null;
+        $this->currentEndpoint = null;
+    }
+
+    public function getQueryCount(): int
+    {
+        return count($this->getQueries());
+    }
+
+    public function getQueries(): array
+    {
+        return array_key_exists('queries', $this->data) ? $this->data['queries'] : [];
+    }
+
+    public function getTotalTimeMs(): float
+    {
+        return array_key_exists('totalTimeMs', $this->data) ? $this->data['totalTimeMs'] : 0.0;
+    }
+
+    public function serialize(): ?string
+    {
+        return serialize($this->data);
+    }
+
+    public function unserialize($serialized): void
+    {
+        $this->data = unserialize($serialized);
+    }
+
+    /**
+     * Plugin init function
+     *
+     * Register event listeners
+     */
+    protected function initPluginType()
+    {
+        /** @var EventDispatcherInterface $dispatcher */
+        $dispatcher = $this->client->getEventDispatcher();
+        if (!in_array($dispatcher, $this->eventDispatchers, true)) {
+            $dispatcher->addListener(SolariumEvents::PRE_EXECUTE_REQUEST, [$this, 'preExecuteRequest'], 1000);
+            $dispatcher->addListener(SolariumEvents::POST_EXECUTE_REQUEST, [$this, 'postExecuteRequest'], -1000);
+            $this->eventDispatchers[] = $dispatcher;
+        }
+    }
+
+    protected function log(
+        SolariumRequest $request,
+        ?SolariumResponse $response,
+        SolariumEndpoint $endpoint,
+        float $durationSec
+    ): void {
+        $requestUri = str_replace(
+            '://localhost:',
+            '://gant-dev-solr.usemarkup.com:',
```

Comment:
This looks sus. Does this need to be configurable?

Reply:
From what I understand this doesn't work right now at all, as localhost is an internal address inside the VM. So if you want to see this from outside the VM, you need an address that is mapped by the VM.
However, this address is not available anywhere in the codebase, as it is set when creating a VM and saved in Nginx configuration only.

So, the address is the same for every Gant dev machine if it used gant-chef to create the VM.
Two situation I see where this may not be correct:
* If the Symfony Profiler was used on environments other than dev in Gant.
* If this system is used in Hunter.

Reply:
I could move it into parameters.yml, if it makes more sense?

Reply:
Fixed in https://github.com/usemarkup/NeedleBundle/pull/151